docs/specifications/tests/DNSSEC-TP/dnssec01.md

```diff
@@ -33,87 +49,164 @@ Algorithm number | Algorithm (or description)
-      6. If no message has been outputted for the DS, then
-         output *[DS_ALGORITHM_OK]*.
-   2. If *Undelegated DS* has at least one DS but none with
-      algorithm value 2 output *[DS_ALGORITHM_MISSING]*.
-   3. End the steps.
-
-2. Create a DS query with DO flag set for the name of the
-   *Child Zone* (*Test Type* is "normal").
-
-3. Retrieve all name server IP addresses for the parent zone of
-   *Child Zone* using [Method1] ("Parent NS IP").
-
-4. For each IP address in *Parent NS IP* do:
-   1. Send the DS query over UDP and collect the response.
-   2. If there is no DNS response, then output *[NO_RESPONSE_DS]*.
-   3. Else, if the RCODE is not NOERROR, then output
-      *[UNEXPECTED_RESPONSE_DS]*.
-   4. Else, go to next IP address if there is no DS in the
-      response.
-   5. Else, extract all DS records from the DNS response
-      ("DS Records").
-   6. For each DS in *DS Records*, do:
-      1. Compare the DS algorithm value with *Algorithm Status*.
-      2. If the algortithm value is 0 then output
-         *[DS_ALGORITHM_NOT_DS]*.
-      3. If algortithm value is 1 then output
-         *[DS_ALGO_SHA1_DEPRECATED]*.
-      4. If algortithm value is 3 then output
-         *[DS_ALGORITHM_DEPRECATED]*.
-      5. If algortithm value is 5-255 then output
-         output *[DS_ALGORITHM_RESERVED]*.
-      6. If no message has been outputted for the DS, then
-         output *[DS_ALGORITHM_OK]*.
-   7. If there was no DS with algorithm value 2 output
-      *[DS_ALGORITHM_MISSING]*.
+* "Undelegated DS" - The DS record or records submitted, undefined unless
+  *Test Type* is undelegated and empty if no DS record has been submitted.
+
+
+## Summary
+
+* At least one DS record must be found, or no further investigation will be done
+  and no messages will be outputted.
+* No messages will be outputted due to errors in the responses from the parent
+  name servers.
+
+Message Tag outputted          | Level   | Arguments                                | Description of when message tag is outputted
+:------------------------------|:--------|:-----------------------------------------|:--------------------------------------------
+DS01_DIGEST_NOT_SUPPORTED_BY_ZM| NOTICE  | ns_ip_list, algo_mnemo, algo_num, keytag | DS Digest cannot be validated by this installation of Zonemaster.
+DS01_DS_ALGO_DEPRECATED        | ERROR   | ns_ip_list, algo_mnemo, algo_num, keytag | The DS digest algorithm is deprecated.
+DS01_DS_ALGO_2_MISSING         | NOTICE  |                                          | Algo 2 (SHA-256) is expected but missing.
+DS01_DS_ALGO_NOT_DS            | ERROR   | ns_ip_list, algo_mnemo, algo_num, keytag | The DS digest algorithm is not for DS.
+DS01_DS_ALGO_RESERVED          | ERROR   | ns_ip_list, algo_mnemo, algo_num, keytag | No DS digest algorithm defined for the digest code.
+DS01_DS_ALGO_SHA1_DEPRECATED   | WARNING | ns_ip_list, algo_mnemo, algo_num, keytag | The DS SHA1 digest algorithm is deprecated.
```

Comment:
Can you confirm that it is justified enough to treat SHA-1 specially? It has a specific message tag for deprecation, with different level, instead of using DS01_DS_ALGO_DEPRECATED

Reply:
2 years ago it was decided that the use of SHA1 was too wide spread to just output a clean ERROR (#839). [IANA] still lists algorithm 1 (SHA-1) as "mandatory". E.g. arpa has still a algorithm 1 DS record along side a algorithm 2 record:
```
;; ANSWER SECTION:
arpa.			54732 IN DS 42581 8 2 (
				F28391C1ED4DC0F151EDD251A3103DCE0B9A5A251ACF
				6E24073771D71F3C40F9 )
arpa.			54732 IN DS 42581 8 1 (
				778606D9623F843F156E7D11ACBF815EB67AB516 )
```

It worth discussing this again for the v2022.2 release.

[IANA]: https://www.iana.org/assignments/ds-rr-types/ds-rr-types.xml